packages/strapi-plugin-upload/admin/src/components/UploadList/index.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable react/no-array-index-key */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { ModalBody } from 'strapi-helper-plugin';
+import { Button } from '@buffetjs/core';
+import createMatrix from '../../utils/createMatrix';
+import getTrad from '../../utils/getTrad';
+import CardImgWrapper from '../CardImgWrapper';
+import InfiniteLoadingIndicator from '../InfiniteLoadingIndicator';
+import HeaderWrapper from './HeaderWrapper';
+
+const UploadList = ({
+  filesToUpload,
+  onClickCancelUpload,
+  onGoToAddBrowseFiles,
+}) => {
+  const matrix = createMatrix(filesToUpload);
+  const filesToUploadLength = filesToUpload.length;
+  const titleId = `modal.upload-list.sub-header-title.${
+    filesToUploadLength > 1 ? 'plural' : 'singular'
+  }`;
+
+  return (
+    <>
+      <HeaderWrapper>
+        <div>
+          <div className="assets-selected">
+            <FormattedMessage
+              id={getTrad(titleId)}
+              values={{ number: filesToUploadLength }}
+            />
+          </div>
+          <div className="infos">
```

Comment:
Would work with the `<Text>` component.  We could also create an `id` prop in the `<Text>` component to display intl messages.